### PR TITLE
Add lazy UploadFormField registration

### PR DESCRIPTION
**Pull request by joshuaruesweg**
If request data is to be transferred to the form, the field may otherwise be initialised with incorrect data.  For references please see the following bug report:  https://community.woltlab.com/thread/288462-uploadformfield-nicht-in-dialogformdocument-m%C3%B6glich/

### CODE REVIEW

File: wcfsetup/install/files/lib/system/form/builder/field/UploadFormField.class.php [hunk 1]
```diff
@@ -100,16 +100,35 @@ class UploadFormField extends AbstractFormField
...
+
     /**
      * Unregisters the current field in the upload handler.
      */
     private function unregisterField()
     {
         if (UploadHandler::getInstance()->isRegisteredFieldId($this->getPrefixedId())) {
             UploadHandler::getInstance()->unregisterUploadField($this->getPrefixedId());
+
+            $this->registered = false;
```
Comment (TimWolla) [inline wcfsetup/install/files/lib/system/form/builder/field/UploadFormField.class.php:118]: s/registered/isRegistered/?

**Review by TimWolla — COMMENTED** [review]
Can't say anything else about this. This is not my area of expertise.

File: wcfsetup/install/files/lib/system/form/builder/field/UploadFormField.class.php [hunk 2]
```diff
@@ -100,16 +100,35 @@ class UploadFormField extends AbstractFormField
...
     {
         if (UploadHandler::getInstance()->isRegisteredFieldId($this->getPrefixedId())) {
             UploadHandler::getInstance()->unregisterUploadField($this->getPrefixedId());
+
+            $this->isRegistered = false;
         }
     }
 
+    /**
+     * Returns true, if the current field is already registered.
```
Comment (Gravatronics) [inline wcfsetup/install/files/lib/system/form/builder/field/UploadFormField.class.php:123]:
```suggestion
     * Returns `true` if the current field has already registered with the upload handler.
```

**Review by Gravatronics — COMMENTED** [review]
Everything else LGTM.

File: wcfsetup/install/files/lib/system/form/builder/field/UploadFormField.class.php (hunk 2 above)
Comment (joshuaruesweg) [reply wcfsetup/install/files/lib/system/form/builder/field/UploadFormField.class.php:123]: Fix grammar in phpDoc comment

**Pull request merged**